src/containers.tsx

```diff
@@ -45,7 +45,7 @@ export function createDisabledContainer (WrappedComponent: React.ComponentClass<
 export function createGuardedContainer ({ isGuarded, enabledComponent, disabledComponent }: IGuardedContainerProps): React.ComponentClass {
   @observer
   class GuardedContainer extends Component {
-    private readonly GuardedComponent: React.ComponentClass;
+    private readonly GuardedComponent: any;
```

Comment:
I know this isn't optimal, but I spend a lot of time trying to figure that one out.
The issue is that we're setting that value after the constructor has triggered, so it would technically need to be :
`private readonly GuardedComponent:  React.ComponentClass<any> | {} | null | undefined;`

Which felt ... messy || not efficient enough to justify it ?

Let me know what you think.